code/game/machinery/computer/law.dm

```diff
@@ -36,12 +49,52 @@
 		if(!atoms_share_level(T, src))
 			to_chat(user, "<span class='danger'>Unable to establish a connection</span>: You're too far away from the target silicon!")
 			return
-		var/obj/item/aiModule/M = O
-		M.install(src)
+		if(!emagged) //non-emag law change
+			var/obj/item/aiModule/M = O
+			M.install(src)
+			return
+		else
+			if(world.time < cooldown)
+				to_chat(user, "The program seems to have frozen. It will need some time to process.")
+				return
+			do_sparks(5, TRUE, src)
+			var/foundlaws = 0
+			var/checked = FALSE
+			for(var/datum/ai_law/law in current.laws.all_laws())
+				if(law in current.laws.inherent_laws)
+					foundlaws++
+				if(checked)
+					continue
+				if(!length(current.laws.ion_laws))
+					checked = TRUE
+					if(prob(20))  // 20% chance to generate an ion law if none exists
+						current.add_ion_law(generate_ion_law())
+						cooldown = world.time + cooldown_delay
+						return
+				else if(law in current.laws.ion_laws) //10% chance to overwrite a current ion
+					checked = TRUE
+					if(prob(10))
+						current.clear_ion_laws()
+						current.add_ion_law()
+						cooldown = world.time + cooldown_delay
+						return
+			var/lawposition = rand(1, foundlaws)
+			if(foundlaws) //as long as it finds a law to change
+				current.laws.inherent_laws[lawposition].law = new/datum/ai_law/inherent(generate_ion_law()).law
+				log_and_message_admins("has given [current] the emag'd inherent law: [current.laws.inherent_laws[lawposition].law].")
+				current.show_laws()
+				current.throw_alert("newlaw", /obj/screen/alert/newlaw)
+				cooldown = world.time + cooldown_delay
+				for(var/i in 1 to length(current.connected_robots)) // push alert to the AI's borgs
+					current.connected_robots[i].cmd_show_laws()
+					current.connected_robots[i].throw_alert("newlaw", /obj/screen/alert/newlaw)
+			return
 		return
 	return ..()
 
 
+/obj/machinery/computer/aiupload/proc/add_emag_law()
+
```

Comment:
Move the code that sirryan selected into here, then you will write in the place where all those lines have been removed "add_emag_law()".